api/src/test/java/io/strimzi/api/kafka/model/KafkaMirrorMaker2CrdIT.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model;
+
+import io.strimzi.test.TestUtils;
+import io.strimzi.test.k8s.exceptions.KubeClusterException;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.CoreMatchers.containsStringIgnoringCase;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.CoreMatchers.allOf;
+import static org.hamcrest.CoreMatchers.anyOf;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+/**
+ * The purpose of this test is to confirm that we can create a
+ * resource from the POJOs, serialize it and create the resource in K8S.
+ * I.e. that such instance resources obtained from POJOs are valid according to the schema
+ * validation done by K8S.
+ */
+public class KafkaMirrorMaker2CrdIT extends AbstractCrdIT {
+    public static final String NAMESPACE = "kafkamirrormaker2-crd-it";
+
+    @Test
+    void testKafkaMirrorMaker2V1alpha1() {
+        assumeKube1_11Plus();
+        createDelete(KafkaMirrorMaker2.class, "KafkaMirrorMaker2V1alpha1.yaml");
+    }
+
+    @Test
+    void testKafkaMirrorMaker2Minimal() {
+        createDelete(KafkaMirrorMaker2.class, "KafkaMirrorMaker2-minimal.yaml");
+    }
+
+    @Test
+    void testKafkaMirrorMaker2WithExtraProperty() {
+        createDelete(KafkaMirrorMaker2.class, "KafkaMirrorMaker2-with-extra-property.yaml");
+    }
+
+    @Test
+    void testKafkaMirrorMaker2WithMissingRequired() {
+        Throwable exception = assertThrows(
+            KubeClusterException.InvalidResource.class,
+            () -> {
+                createDelete(KafkaMirrorMaker2.class, "KafkaMirrorMaker2-with-missing-required-property.yaml");
+            });
+
+
+        assertThat(exception.getMessage(), anyOf(
+                allOf(
+                    containsStringIgnoringCase("spec.connectCluster in body is required"),
+                    containsStringIgnoringCase("spec.clusters.alias in body is required"),
```

Comment:
I think depending on the version of kube you're using you can have messages like 

    containsStringIgnoringCase("spec.bootstrapServers in body is required"),

or 

    containsStringIgnoringCase("spec.bootstrapServers: Required value")));

Now might be a good time to introduce a helper method to `AbstractCrdIT` for asserting the required message for a given path (you don't have to update all the existing `CrdIT`s, that can be done in a separate commit).

Reply:
Done